python/lsst/ts/mthexapod/hexapod_csc.py

```diff
@@ -629,6 +662,62 @@ async def do_move(self, data: salobj.BaseMsgType) -> None:
         )
         await self.move_task
 
+    def get_filter_z_offset(self, filter_name: str) -> float:
+        """Get the z offset for a given filter name.
+
+        Parameters
+        ----------
+        filter_name : `str`
+            Camera filter name.
+
+        Returns
+        -------
+        z_offset : `float`
+            Z offset for the filter name.
+            Returns 0 if filter name not found or filter_offsets empty.
+        """
+        if not self.filter_offsets:
+            return 0
+
+        for filter in self.filter_offsets:
```

Comment:
beware that `filter` is a python keyword (like `return` or `for`):

```
In [1]: filter?
Init signature: filter(self, /, *args, **kwargs)
Docstring:
filter(function or None, iterable) --> filter object

Return an iterator yielding those items of iterable for which function(item)
is true. If function is None, return the items that are true.
Type:           type
Subclasses:
```

so this will shadow `filter` in this context. Please consider renaming it (here and elsewhere).